cmake/Tools/Platform/Android/generate_android_project.py

```diff
@@ -126,6 +126,8 @@ def build_optional_signing_profile(store_file, store_password, key_alias, key_pa
 ASSET_TYPE_ARGUMENT_NAME = '--asset-type'
 DEFAULT_ASSET_TYPE = 'android'
 
+DEFAULT_3RD_PARTY_PATH = str(pathlib.Path(os.path.expanduser('~')) / '.o3de' / '3rdParty')
```

Comment:
We should read this value from the o3de_manifest.json via using importing manifest.py module into generate_android_project.py.

The o3de_manifest.json contains a "default_third_party_folder" key which should be read.
Otherwise we should fallback to invoking the `manifest.get_o3de_third_party_folder` method to return a default 3rdParty directory.

```
from o3de import manifest
# ...
json_data = manifest.load_o3de_manifest()

DEFAULT_3RD_PARTY_PATH = json_data.get('default_third_party_folder', manifest.get_o3de_third_party_folder())
```